packages/modules/imodel-browser/src/containers/iModelGrid/IModelGrid.tsx

```diff
@@ -120,53 +159,193 @@ export const IModelGrid = ({
+                name="searchText"
+                onChange={handleChange}
+                value={searchText}
+                iconDisplayStyle="inline"
+                svgIcon={
+                  searchText?.trim() ? (
+                    <div ref={closeRef} onClick={() => setSearchText("")}>
+                      <SvgClose />
+                    </div>
+                  ) : (
+                    <div ref={searchRef}>
+                      <SvgSearch />
+                    </div>
+                  )
+                }
               />
-            ))}
-            {fetchMore ? (
+              <Tooltip reference={searchRef} content="Search" />
+              <ButtonGroup className="iac-view-toolbar">
+                <IconButton
+                  onClick={() => setView("tileView")}
+                  isActive={view === "tileView"}
+                >
+                  <SvgThumbnails />
+                </IconButton>
+                <IconButton
+                  onClick={() => setView("listView")}
+                  isActive={view === "listView"}
+                >
+                  <SvgList />
+                </IconButton>
+              </ButtonGroup>
+            </div>
+          </div>
+        )}
+        {view === "tileView" ? (
+          <GridStructure>
+            {fetchStatus === DataStatus.Fetching ? (
               <>
-                <InView onChange={fetchMore}>
-                  <IModelGhostTile />
-                </InView>
+                <IModelGhostTile />
                 <IModelGhostTile />
                 <IModelGhostTile />
               </>
-            ) : null}
-          </>
+            ) : (
+              <>
+                {searchText && iModels.length === 0 ? (
+                  <div>No Results found for search</div>
+                ) : (
+                  iModels?.map((iModel) => (
+                    <IModelHookedTile
+                      key={iModel.id}
+                      iModel={iModel}
+                      iModelOptions={iModelActions}
+                      accessToken={accessToken}
+                      onThumbnailClick={onThumbnailClick}
+                      apiOverrides={tileApiOverrides}
+                      useTileState={useIndividualState}
+                      {...tileOverrides}
+                    />
+                  ))
+                )}
+                {fetchMore ? (
+                  <>
+                    <InView onChange={fetchMore}>
+                      <IModelGhostTile />
+                    </InView>
+                    <IModelGhostTile />
+                    <IModelGhostTile />
+                  </>
+                ) : null}
+              </>
+            )}
+          </GridStructure>
+        ) : (
+          <Table<{ [P in keyof IModelFull]: IModelFull[P] }>
+            columns={columns}
+            data={data}
+            onRowClick={onRowClick}
+            emptyTableContent="No data."
+            isSortable
```

Comment:
Yes, we just need to give table sorting once it is been fetched.

Reply:
But if we fetch the 100 first iModels, then invert the sort, we will not fetch the  "last 100" but only reorder what is already fetched... The sort order must be passed down to the `useImodelData` so the correct data is downloaded.
(That issue will be even more visible with the 10 by 10 that we are using, it will only sort the 10 visible, not the top 10 of all users iModels for the iTwin)